cache/network/download.go

```diff
@@ -39,7 +39,6 @@ func Download(ctx context.Context, params DownloadParams, logger log.Logger) (ma
 	}
 
 	retryableHTTPClient := retryhttp.NewClient(logger)
-	retryableHTTPClient.CheckRetry = customRetryFunction
```

Comment:
Can `customRetryFunction` be removed?

Reply:
It could be. I kept it mainly because of the related test, which shows when it'll do a retry and when it won't.

Reply:
It's not a blocker, but let's continue this discussion as we're now testing an unused method and I don't understand its value

Reply:
Actually you're right, we don't need the `customRetryFunction` func anymore. Kept the tests, but those no longer depend on the `customRetryFunction`, so I removed that func.

Reply:
I moved `customRetryFunction` to the test file, as the implementation doesn't depend on it. For now I'd keep the test for future testing but we might remove it eventually completely.